traits/trait_handlers.py

```diff
@@ -2757,6 +2757,7 @@ def update ( self, value ):
         if not hasattr(self, 'trait'):
             return set.update(self, value)
         try:
+            value = set(value)
```

Comment:
@cfarrow: This is 'just' a performance worry, right? I think this is probably fine for now.